clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -1328,7 +1328,7 @@ Priority priority() {
         @Override
         public void handleResponse(AbstractResponse response) {
             AddPartitionsToTxnResponse addPartitionsToTxnResponse = (AddPartitionsToTxnResponse) response;
-            Map<TopicPartition, Errors> errors = addPartitionsToTxnResponse.errors();
+            Map<TopicPartition, Errors> errors = addPartitionsToTxnResponse.errors().get(AddPartitionsToTxnResponse.V3_AND_BELOW_TXN_ID);
```

Comment:
What should we do if the check fails? Just have a better error message thrown?

Reply:
Yeah, the check is probably not necessary. By the way, I find the idea of having `V3_AND_BELOW_TXN_ID` for old version a bit confusing. I was wondering if using `addPartitionsToTxnResponse.data().resultsByTopicV3AndBelow()` would be a better alternative here. We only iterate over the Map so the Map is not strictly required here. Have you considered something like this?

Reply:
I was told not to have v3 and below specific methods from Jason because the v3 case should generalize to a single version of the v4 case and that should make it easy to use methods for both.

However, if we really think this is an issue. I guess we can change the approach again. I'm just not sure the experience of errors only applying to v4+. Any ideas there besides changing the method name to express it should only be used in v4+?

Reply:
Understood. Let's keep it as it is then.

I agree that v3 case should generalized to a single item of the v4 case. It is just unfortunate that we don't have the transaction id in v3 response so we have to use an empty string for it. I suppose that it is the way it is.

Reply:
Yeah. It really is unfortunate. 😞